cmd/oras/login.go

```diff
@@ -20,36 +20,33 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"io"
 	"os"
 	"strings"
 
-	"github.com/moby/term"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 
+	"github.com/moby/term"
 	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras/cmd/oras/internal/option"
 	"oras.land/oras/internal/credential"
 	"oras.land/oras/internal/http"
 	"oras.land/oras/internal/trace"
 )
 
 type loginOptions struct {
-	hostname  string
-	fromStdin bool
-
-	debug      bool
-	configs    []string
-	caFilePath string
-	username   string
-	password   string
-	insecure   bool
-	plainHTTP  bool
-	verbose    bool
+	*option.Common
+	*option.Credential
+	*option.TLS
```

Comment:
It would be better to have values instead of pointers.
```suggestion
	option.Common
	option.Credential
	option.TLS
```

Reply:
It can't be a struct. If I use value here, then 
- the receiver argument of `ApplyFlagsTo` function for common/credential/tls options need to be a struct (otherwise loginOptions.Common is not an `applier` interface)
- then the `ApplyFlagsTo` method will be called on a copied struct, and after the method returns options will still be default value